src/domains/collections.ts

```diff
@@ -0,0 +1,256 @@
+    return orderedLines.reduce(
+      (box: SpatialExtent, line: any) => mergeBoxes(box, line),
+      null as SpatialExtent
+    );
+  }
+
+  if (collection.boxes) {
+    return collection.boxes.reduce(
+      (box: SpatialExtent, boxStr: string) =>
+        mergeBoxes(box, reorderBoxValues(parseOrdinateString(boxStr))),
+      []
+    );
+  }
+
+  return WHOLE_WORLD_BBOX as SpatialExtent;
+};
+
+const createExtent = (collection: Collection): Extents => {
+  return {
+    spatial: {
+      bbox: [cmrCollSpatialToExtents(collection)],
+    },
+    temporal: {
+      interval: [[collection.timeStart, collection.timeEnd]],
+    },
+  };
+};
+
+export const collectionToStac = (collection: any): STACCollection => {
+  const extent = createExtent(collection);
+  const { directDistributionInformation } = collection;
+
+  let assets: AssetLinks = {};
+  if (directDistributionInformation) {
+    assets = directDistributionInformation.s3BucketAndObjectPrefixNames
+      // this handles badly formatted lists of links intead of correct array of strings
+      .flatMap((s3Link: string) => s3Link.split(","))
+      .map((s3Link: string) => s3Link.trim())
+      .filter((s3Link: string) => s3Link.startsWith("s3://"))
+      .reduce((acc: AssetLinks, href: string) => {
+        const assetTitle = href
+          .replace("s3://", "")
+          .replace(/[\/\-:\.]/gi, "_");
+        const newAsset: AssetLinks = {};
+        newAsset[`s3_${assetTitle}`] = { href, roles: ["data"] };
+        return { ...acc, ...newAsset };
+      }, {});
+  }
+
+  return {
+    type: "Collection",
+    id: collection.conceptId,
+    title: collection.title,
+    description: collection["abstract"],
+    license: collection.useConstraints?.description ?? "not-provided",
+    stac_version: "1.0.0",
+    extent,
+    assets,
+    links: [
+      {
+        rel: "about",
+        href: `${CMR_ROOT}/search/concepts/${collection.conceptId}.html`,
+        title: "HTML metadata for collection",
+        type: "text/html",
+      },
+      {
+        rel: "via",
+        href: `${CMR_ROOT}/search/concepts/${collection.conceptId}.json`,
+        title: "CMR JSON metadata for collection",
+        type: "application/json",
+      },
+      {
+        rel: "via",
+        href: `${CMR_ROOT}/search/concepts/${collection.conceptId}.umm_json`,
+        title: "CMR UMM_JSON metadata for collection",
+        type: "application/vnd.nasa.cmr.umm+json",
+      },
+    ],
+  };
+};
+
+export const getCollections = async (
+  query: CollectionsInput,
+  opts: {
+    headers?: { "client-id"?: string; [key: string]: any };
+    [key: string]: any;
+  } = {}
+): Promise<{
+  count: number;
+  facets: FacetGroup | null;
+  cursor: string | null;
+  items: STACCollection[];
+}> => {
+  let userClientId = "cmr-stac";
+  let authorization;
+
+  const { headers } = opts;
+  if (headers) {
+    userClientId = headers["client-id"]
+      ? `${headers["client-id"]}-cmr-stac`
```

Comment:
Is `cmr-stac` enough for the system's client-id? @abbottry has been working to get all the client ids more uniform

Also, I found this chunk of code I think 3 different times, great usecase for a util method

Reply:
It is mentioned in the AC of the ticket, you can find the detailed explanation in EDSC-3587 as well which was the ticket that resulted in this code https://github.com/nasa/cmr-graphql/blob/main/src/graphql/handler.js#L83